### PR TITLE
[docs] Add videos for universal app tutorial

### PR DESCRIPTION
**Pull request by betomoedano**
# Why

Fixes [DES-92](https://linear.app/expo/issue/DES-92/[video]-tutorial-video-series)

# How

Used `VideoBoxLink` to embed videos in each lesson

# Test Plan

Run locally and validate videos display correctly.

https://github.com/user-attachments/assets/906d3257-4840-4317-af38-f5d8b1c223c9

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by amandeepmittal — CHANGES_REQUESTED** [review]
A few things we can improve to avoid redundancy and make these links more actionable.

- Remove descriptions from each VideoBoxLink since they are redundant because the these video boxes are added just after the introduction. The introduction and page description already has description to explain the same.

CleanShot 2024-10-10 at 17 02 55@2x

- Instead of copying/pasting the YT video title, let's make it more meaningful and actionable by adding the prefix "Watch: ". Here's an example for Create your first app chapter:

CleanShot 2024-10-10 at 17 04 41@2x

Notice that the title is not redundant and is more actionable. It is also grammatically correct and makes sense.

CleanShot 2024-10-10 at 17 07 41@2x

This will also allow the text for each boxlink in the tutorial to be parallel which is generally appreciated by readers as it makes the content scannable.

Another example:

CleanShot 2024-10-10 at 17 10 21@2x

File: docs/pages/tutorial/add-navigation.mdx [hunk 1]
```diff
@@ -9,9 +9,16 @@ import { FileTree } from '~/ui/components/FileTree';
 import { ProgressTracker } from '~/ui/components/ProgressTracker';
 import { Step } from '~/ui/components/Step';
 import { CODE } from '~/ui/components/Text';
+import { VideoBoxLink } from '~/ui/components/VideoBoxLink';
 
 In this chapter, we'll learn Expo Router's fundamentals to create stack navigation and a bottom tab bar with two tabs.
 
+<VideoBoxLink
+  videoId="8336fcFV_T4"
```
Comment (Simek) [inline docs/pages/tutorial/add-navigation.mdx:17]: This video id is used twice, on different pages, with different metadata. Is it a mistake?

Comment (betomoedano) [reply docs/pages/tutorial/add-navigation.mdx:17]: Good catch! 😅

**Review by amandeepmittal — APPROVED** [review]
Please see my comment about the title on "create a modal" box link and let's update it to align with others. Overall, all changes look good to me. 

Thank you!

**Review by Simek — APPROVED** [review]
Thanks for working on this! 🚀

**Pull request merged**